sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -154,9 +154,12 @@ object FileSourceStrategy extends Strategy with PredicateHelper with Logging {
         l.resolve(
           fsRelation.partitionSchema, fsRelation.sparkSession.sessionState.analyzer.resolver)
       val partitionSet = AttributeSet(partitionColumns)
-      val partitionKeyFilters =
+      val partitionKeyFilters = if (partitionColumns.isEmpty) {
+        ExpressionSet(Nil)
+      } else {
         ExpressionSet(normalizedFilters
           .filter(_.references.subsetOf(partitionSet)))
+      }
 
       logInfo(s"Pruning directories with: ${partitionKeyFilters.mkString(",")}")
```

Comment:
nit: Is this logging still meaningful for the non-partition case? I mean, could we move it inside the `else` block in the line 159-162?

Reply:
makes sense